app/experimenter/nimbus-ui/src/components/PageEditAudience/FormAudience/index.test.tsx

```diff
@@ -110,6 +113,152 @@ describe("FormAudience", () => {
+          ],
+          isSticky: true,
+        }}
+        config={{
+          ...MOCK_CONFIG,
+          targetingConfigs: [
+            {
+              label: "No Targeting",
+              value: "",
+              applicationValues: [
+                NimbusExperimentApplicationEnum.DESKTOP,
+                "TOASTER",
+              ],
+              description: "No Targeting configuration",
+              stickyRequired: false,
+            },
+            {
+              label: "Win Only",
+              value: "WIN_ONLY",
+              applicationValues: [NimbusExperimentApplicationEnum.DESKTOP],
+              description: "Win Only configuration",
+              stickyRequired: true,
+            },
+            {
+              label: "Toaster thing",
+              value: "TOASTER_THING",
+              applicationValues: ["TOASTER"],
+              description: "Toaster thing description",
+              stickyRequired: false,
+            },
+          ],
+        }}
+      />,
+    );
+
+    const targetingConfigSlug = (await screen.findByTestId(
+      "targetingConfigSlug",
+    )) as HTMLSelectElement;
+
+    expect(targetingConfigSlug.value).toEqual(
+      MOCK_CONFIG!.targetingConfigs![1]!.value,
+    );
+    // since sticky is required for the selected targeting
+    expect(screen.getByTestId("isSticky")).toHaveProperty("checked", true);
+    expect(screen.getByTestId("isSticky")).toBeDisabled();
+    await expect(
+      screen.getByTestId("stickyRequiredWarning"),
+    ).toBeInTheDocument();
+    // changing the selected targeting to sticky required false
+    fireEvent.change(screen.getByTestId("targetingConfigSlug"), {
+      target: { value: MOCK_CONFIG!.targetingConfigs![0]!.value },
+    });
+    // since sticky is not required for the selected targeting
+    expect(screen.getByTestId("isSticky")).toHaveProperty("checked", false);
+    expect(screen.getByTestId("isSticky")).not.toBeDisabled();
+    await expect(
+      screen.queryByTestId("stickyRequiredWarning"),
+    ).not.toBeInTheDocument();
+    // Making sticky enrollment for not sticky required targeting
+    fireEvent.click(screen.getByTestId("isSticky"), {
+      target: { checked: false },
+    });
+    // since sticky is not required, user clicked to have it sticky
+    expect(screen.getByTestId("isSticky")).toHaveProperty("checked", true);
+    expect(screen.getByTestId("isSticky")).not.toBeDisabled();
+    await expect(
+      screen.queryByTestId("stickyRequiredWarning"),
+    ).not.toBeInTheDocument();
+    // selecting targeting which requires stick enrollment
+    fireEvent.change(screen.getByTestId("targetingConfigSlug"), {
+      target: { value: MOCK_CONFIG!.targetingConfigs![1]!.value },
+    });
+    expect(screen.getByTestId("isSticky")).toHaveProperty("checked", true);
+    expect(screen.getByTestId("isSticky")).toBeDisabled();
+    await expect(
+      screen.getByTestId("stickyRequiredWarning"),
+    ).toBeInTheDocument();
+  });
```

Comment:
These are all the right cases but I would split each of these into separate tests rather than putting them all together in one.  Testing too many things in one test can make it hard to know which specific thing is broken when the test fails since then you have to go digging around at which line is failing its assertion rather than just looking at the test name itself.